src/providers/wms/qgswmsprovider.cpp

```diff
@@ -1701,7 +1701,7 @@ void QgsWmsProvider::setupXyzCapabilities( const QString &uri, const QgsRectangl
     tm.topLeft = topLeft;
     tm.tileWidth = tm.tileHeight = 256 * tilePixelRatio;
     tm.matrixWidth = tm.matrixHeight = 1 << zoom;
-    tm.tres = xspan / ( tm.tileWidth * tm.matrixWidth );
+    tm.tres = xspan / ( ( double ) tm.tileWidth * tm.matrixWidth );
```

Comment:
Does this work instead?

```suggestion
    tm.tres = xspan / ( static_cast< qgssize >( tm.tileWidth ) * tm.matrixWidth );
```

Reply:
Yes, it works.